broker.go

```diff
@@ -338,6 +371,8 @@ func (b *Broker) send(rb protocolBody, promiseResponse bool) (*responsePromise,
 		return nil, err
 	}
 
+	b.updateOutgoingCommunicationMetrics(len(buf))
```

Comment:
probably want to move this down below the `SetWriteDeadline` else you could miscount in the (admittedly exceptional) case where that fails